### ISSUE TITLE
[Proto] Support Cosmos scalar type definitions

### ISSUE THREAD

**Issue by byte-bandit** [VolumeFi/paloma#1211]
CosmosSDK released support for scalar type definitions: https://github.com/cosmos/cosmos-sdk/pull/9933

They make working with custom types much easier for us, and support for other languages than `Go` isn't needed in our case. The gravity module already makes use of them, but our compiler doesn't actually apply the needed changes, meaning the underlying datatypes are still native typed.

We should upgrade this some time.

Opening this issue to keep track of it.

**Comment by maharifu** [comment]
Is the implementation of the scalar types really finished, though? It seems it adds the scalar types annotations, but doesn't change the generators, meaning we still get the underlying native types. That PR does not change the resulting data types.

For instance, the latest version of cosmos still has [string address fields](https://github.com/cosmos/cosmos-sdk/blob/main/x/consensus/types/tx.pb.go#L38) even though they have the scalar type annotation in the [proto definition](https://github.com/cosmos/cosmos-sdk/blob/main/x/consensus/proto/cosmos/consensus/v1/tx.proto#L33). I compiled the protos from the HEAD of the repo and got the same results. Upgrading our proto compiler also yields the same.

We could still add the annotations, but we don't seem to get much benefit from it now.

**Comment by byte-bandit** [comment]
Nah, this was really meant as a tracking ticket so I wouldn't forget about this. I don't think they'll update their generators any time soon tbh. Feel free to skip this one.
